runtime/src/accounts_db.rs

```diff
@@ -1397,8 +1426,52 @@ impl AccountsDB {
         hashes
     }
 
+    pub fn freeze_accounts(&self, ancestors: &HashMap<Slot, usize>, account_pubkeys: &[Pubkey]) {
+        let mut frozen_accounts = self.frozen_accounts.lock().unwrap();
+        for account_pubkey in account_pubkeys {
+            let (account, _slot) = self
+                .load_slow(ancestors, &account_pubkey)
+                .unwrap_or_default();
+            let frozen_account_info = FrozenAccountInfo {
+                hash: Self::hash_frozen_account_data(&account),
+                lamports: account.lamports,
+            };
+            warn!(
+                "Account {} is now frozen at lamports={}, hash={}",
+                account_pubkey, frozen_account_info.lamports, frozen_account_info.hash
+            );
+            frozen_accounts.insert(*account_pubkey, frozen_account_info);
+        }
+    }
+
     /// Store the account update.
     pub fn store(&self, slot_id: Slot, accounts: &[(&Pubkey, &Account)]) {
+        let mut frozen_accounts = self.frozen_accounts.lock().unwrap();
```

Comment:
We take the `frozen_accounts` lock here, I suspect it's not going to be a bottleneck because we're about to do disk I/O anyway.  But I have no data to back up this up.

Reply:
It matters a lot, this is our main bottleneck in accounts